### PR TITLE
MS Teams Upload attachment feature 

### PR DESCRIPTION
**Pull request by vaibhav-db**

### CODE REVIEW

File: libs/chat-workflow/src/main/java/org/finos/springbot/workflow/java/resolvers/ChatButtonWorkflowResolverFactory.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package org.finos.springbot.workflow.java.resolvers;
+
+import java.util.Optional;
+
+import org.finos.springbot.workflow.annotations.ChatButton;
+import org.finos.springbot.workflow.java.mapping.ChatHandlerExecutor;
+
+/**
+ * Get chat button detail in controller
```
Comment (robmoffat) [inline libs/chat-workflow/src/main/java/org/finos/springbot/workflow/java/resolvers/ChatButtonWorkflowResolverFactory.java:9]: Can you describe what this is for?

File: libs/teams/teams-chat-workflow-spring-boot-starter/src/main/java/org/finos/springbot/teams/messages/FileActivityHandler.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+import com.microsoft.bot.builder.MessageFactory;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.Attachment;
+import com.microsoft.bot.schema.ResultPair;
+import com.microsoft.bot.schema.TextFormatTypes;
+import com.microsoft.bot.schema.teams.FileConsentCardResponse;
+import com.microsoft.bot.schema.teams.FileInfoCard;
+
+public class FileActivityHandler extends MessageActivityHandler {
```
Comment (robmoffat) [inline libs/teams/teams-chat-workflow-spring-boot-starter/src/main/java/org/finos/springbot/teams/messages/FileActivityHandler.java:34]: Please check the code coverage as a result of adding this.  I have a feeling there will need to be some unit tests added to make sure this works.

**Review by robmoffat — CHANGES_REQUESTED** [review]
Looking good so far!

File: libs/chat-workflow/src/main/java/org/finos/springbot/workflow/java/resolvers/ChatButtonWorkflowResolverFactory.java (hunk 1 above)
Comment (vaibhav-db) [reply libs/chat-workflow/src/main/java/org/finos/springbot/workflow/java/resolvers/ChatButtonWorkflowResolverFactory.java:9]: As we discussed last time, we need button information on controller method. For that we have added this class.

File: libs/teams/teams-chat-workflow-spring-boot-starter/src/main/java/org/finos/springbot/teams/messages/FileActivityHandler.java (hunk 2 above)
Comment (vaibhav-db) [reply libs/teams/teams-chat-workflow-spring-boot-starter/src/main/java/org/finos/springbot/teams/messages/FileActivityHandler.java:34]: Sure, I will check the code coverage and also add the unit testcases.

Comment (vaibhav-db) [reply libs/teams/teams-chat-workflow-spring-boot-starter/src/main/java/org/finos/springbot/teams/messages/FileActivityHandler.java:34]: committed testcases..

Comment (vaibhav-db) [reply libs/teams/teams-chat-workflow-spring-boot-starter/src/main/java/org/finos/springbot/teams/messages/FileActivityHandler.java:34]: Hi @robmoffat ,

Now code coverage is 72.53%

Comment (robmoffat) [reply libs/teams/teams-chat-workflow-spring-boot-starter/src/main/java/org/finos/springbot/teams/messages/FileActivityHandler.java:34]: Take a look at this one: https://app.codecov.io/gh/finos/spring-bot/blob/spring-bot-develop/libs%2Fteams%2Fteams-chat-workflow-spring-boot-starter%2Fsrc%2Fmain%2Fjava%2Forg%2Ffinos%2Fspringbot%2Fteams%2Fmessages%2FFileActivityHandler.java

**Review by robmoffat — APPROVED** [review]

**Pull request closed without merging**